app/src/main/java/com/arctouch/codechallenge/di/component/AppComponent.kt

```diff
@@ -0,0 +1,29 @@
+package com.arctouch.codechallenge.di.component
+
+import android.content.Context
+import com.arctouch.codechallenge.MoviesApplication
+import com.arctouch.codechallenge.detail.DetailViewModel
+import com.arctouch.codechallenge.di.ViewModelFactory
+import com.arctouch.codechallenge.di.module.AppModule
+import com.arctouch.codechallenge.home.HomeViewModel
+import dagger.BindsInstance
+import dagger.Component
+import javax.inject.Singleton
+
+@Singleton
+@Component(modules = [AppModule::class])
```

Comment:
A more concise version for building `@Component`s and `@Module`s with `@Binds`, `@ContributesAndroidInjector` can be achieved by using Dagger Android library, have you ever tried it? You can read more about it at this link: https://proandroiddev.com/dagger-2-annotations-binds-contributesandroidinjector-a09e6a57758f